ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -121,6 +125,36 @@ def create_tf_policy(
         Create a Policy object that uses the TensorFlow backend.
         """
         pass
+    
+    def create_saver(self, policy: Policy) -> Saver:
+        if self.framework == "torch":
+            return self.create_torch_saver(policy)
+        else:
+            return self.create_tf_saver(policy)
+
```

Comment:
I would prefer all these methods to be static on the `Saver` class. Or use a factory (until TF deprecation).